tests/BIAnalyticsService/RetrieveDataTests.cs

```diff
@@ -0,0 +1,132 @@
+using Microsoft.Azure.Functions.Worker.Http;
+using Microsoft.Extensions.Logging;
+using Moq;
+using NHS.ServiceInsights.BIAnalyticsService;
+using NHS.ServiceInsights.TestUtils;
+using System.Collections.Specialized;
+using NHS.ServiceInsights.Common;
+using System.Text;
+
+namespace NHS.ServiceInsights.Tests;
+[TestClass]
+public class RetrieveDataTests
+{
+    private Mock<ILogger<RetrieveData>> _mockLogger = new();
+    private Mock<IHttpRequestService> mock_httpRequestService = new();
+    private RetrieveData _function;
+    private Mock<HttpRequestData> _mockRequest = new();
+    private SetupRequest _setupRequest = new();
+
+    public RetrieveDataTests()
+    {
+
+        Environment.SetEnvironmentVariable("GetEpisodeUrl", "http://localhost:6060/api/GetEpisode");
+        Environment.SetEnvironmentVariable("GetParticipantUrl", "http://localhost:6061/api/GetParticipant");
+        _function = new RetrieveData(_mockLogger.Object, mock_httpRequestService.Object);
+    }
+
+    [TestMethod]
+    public async Task Run_ShouldReturnBadRequest_WhenEpisodeIdIsNotProvided()
+    {
+        // Arrange
+        var queryParam = new NameValueCollection
+        {
+            { "EpisodeId", null }
+        };
+
+        _mockRequest = _setupRequest.SetupGet(queryParam);
+
+        // Act
+        var response = await _function.Run(_mockRequest.Object);
+
+        // Assert
+        Assert.AreEqual(HttpStatusCode.BadRequest, response.StatusCode);
+        _mockLogger.Verify(log =>
+            log.Log(
+            LogLevel.Error,
+            0,
+            It.Is<It.IsAnyType>((state, type) => state.ToString() == "Please enter a valid Episode ID."),
+            null,
+            (Func<object, Exception, string>)It.IsAny<object>()),
+            Times.Once);
+    }
+
+    [TestMethod]
+    public async Task Run_ShouldReturnInternalServerError_WhenExceptionIsThrown()
+    {
+        // Arrange
+        var queryParam = new NameValueCollection
+        {
+            { "episodeId", "745396" }
+        };
+        _mockRequest = _setupRequest.SetupGet(queryParam);
+
+        var url = "http://localhost:6060/api/GetEpisode/?EpisodeId=745396";
+
+        mock_httpRequestService
+            .Setup(service => service.SendGet(url))
+            .ThrowsAsync(new HttpRequestException("Exception: System.Net.Http.HttpRequestException:"));
+
+        // Act
+        var response = await _function.Run(_mockRequest.Object);
+
+        // Assert
+        Assert.AreEqual(HttpStatusCode.InternalServerError, response.StatusCode);
+        _mockLogger.Verify(log => log.Log(
+            LogLevel.Error,
+            0,
+            It.Is<It.IsAnyType>((state, type) => state.ToString().Contains("Failed to call the GetEpisode Data Service.")),
+            null,
+            (Func<object, Exception, string>)It.IsAny<object>()),
+            Times.Once);
+    }
+
+
+    [TestMethod]
+    public async Task RetrieveData_ShouldRetrieveDataFromDownstreamFunctions()
```

Comment:
This test could do with being renamed to more accurately describe what is being asserted